go/signedexchange/signedexchange_test.go

```diff
@@ -389,6 +394,29 @@ func TestVerifyBadSignature(t *testing.T) {
 	})
 }
 
+func TestVerifyNoContentType(t *testing.T) {
+	testForEachVersion(t, func(ver version.Version, t *testing.T) {
+		e, s, c := createTestExchange(ver, t)
+		e.ResponseHeaders.Del("Content-Type");
+		if err := e.AddSignatureHeader(s); err != nil {
+			t.Fatal(err)
+		}
+		certFetcher := func(_ string) ([]byte, error) { return c, nil }
+		verificationTime := signatureDate
+
+		// The requirement for Content-Type is only for version >= b3.
+		if ver == version.Version1b1 || ver == version.Version1b2 {
+			if _, ok := e.Verify(verificationTime, certFetcher, stdoutLogger); !ok {
+				t.Errorf("Verification should succeed")
+			}
+		} else {
+			if _, ok := e.Verify(verificationTime, certFetcher, nullLogger); ok {
```

Comment:
Because I don't want to mess up test output when failure is expected (that's why I use `nullLogger` here), while log message could be useful for debug when failure is unexpected (so `stdoutLogger` in the above).

BTW probably we should use helper functions for these assertions. https://github.com/WICG/webpackage/pull/385